db/system_keyspace.cc

```diff
@@ -3729,13 +3729,13 @@ system_keyspace::read_cdc_generation(utils::UUID id) {
 future<> system_keyspace::sstables_registry_create_entry(sstring location, utils::UUID uuid, sstring status, sstables::entry_descriptor desc) {
     static const auto req = format("INSERT INTO system.{} (location, generation, uuid, status, version, format) VALUES (?, ?, ?, ?, ?, ?)", SSTABLES_REGISTRY);
     slogger.trace("Inserting {}.{}:{} into {}", location, desc.generation, uuid, SSTABLES_REGISTRY);
-    co_await execute_cql(req, location, desc.generation.value(), uuid, status, fmt::to_string(desc.version), fmt::to_string(desc.format)).discard_result();
+    co_await execute_cql(req, location, desc.generation, uuid, status, fmt::to_string(desc.version), fmt::to_string(desc.format)).discard_result();
```

Comment:
Isn't this a little dangerous? `SSTABLES_REGISTRY` defines `generation` as `long_type`, so this will work for now, but as soon as we switch to `UUID` based generations, this will sometimes work and sometimes not. Maybe we can deal with this later.

Reply:
yeah, i just found where `SSTABLES_REGISTRY` table's schema is created. so we cannot have UUID and bigint as its "generation" column. just had a short discussion with @bhalevy , he suggested that we could use string for this column, so we can accommodate the use case of bigint based generation identifier and UUID based one?

Reply:
If we use a string, we are again left with heuristics to try to deduce the real type of the generation.
Maybe we can use two fields, one being a flag denoting the generation type.
AFAIK this table is not in any release yet, so we can still make changes, right @xemul?

Reply:
> AFAIK this table is not in any release yet, so we can still make changes, right @xemul?

The table is under experimental feature and doesn't appear on system keyspace by default, so yes, we can change it the way we want

Reply:
> If we use a string, we are again left with heuristics to try to deduce the real type of the generation.
> Maybe we can use two fields, one being a flag denoting the generation type.
> AFAIK this table is not in any release yet, so we can still make changes, right @xemul?

i think if the ownership table only takes generation as a part of the file / object name , probably there is no need to preserve the actual "type" of the generation id, as what it expects is but a string, right?

Reply:
@denesb @xemul @bhalevy  if no objections, i will change the `generation` column to `ascii_type`. as i think this is the simpler way forwards.

Reply:
Can we change it into `uuid_type` instead? And have some hacky way of wrapping/unwrapping the current integer generation into that for the transition period?

I would prefer a hacky transition and a native type long term, than a solution which makes the transition easier but will be a wart long term.

Reply:
We can generate a time uuid from the integer value and it should be reversible.

Reply:
sure. will 

1. "abuse" the `generation_type::operator data_value` to convert the `generation_type` to uuid even if it is an integer
2. change `generation` column to `uuid`, and let it use the `operator data_value` instead of using `value()` function.